### PR TITLE
Fix problem with complex values injected through env var will broke HOCON config parser

### PR DESCRIPTION
**Pull request by Arkatufus**
Any values that needs to be quoted in HOCON ( containing any character in the set [$"{}[]:=,#`^?!@*&\] and the new line character) will break the HOCON parser and generates an unchecked invalid HOCON config that can cause unpredictable errors in Akka.

### CODE REVIEW

File: src/Akka.Bootstrap.Docker/StringExtension.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Akka.Bootstrap.Docker
+{
+    public static class StringExtension
+    {
+        public const string NotInUnquotedText = "$\"{}[]:=,#`^?!@*&\\";
+        public const char NewLine = '\u000A';
```
Comment (Aaronontheweb) [inline src/Akka.Bootstrap.Docker/StringExtension.cs:11]: Shouldn't this be [`Environment.NewLine`](https://docs.microsoft.com/en-us/dotnet/api/system.environment.newline?view=net-5.0)?

**Review by Aaronontheweb — COMMENTED** [review]
Left one comment

File: src/Akka.Bootstrap.Docker/StringExtension.cs (hunk 1 above)
Comment (Arkatufus) [reply src/Akka.Bootstrap.Docker/StringExtension.cs:11]: Nope, HOCON is very specific on what a newline is

Comment (Arkatufus) [reply src/Akka.Bootstrap.Docker/StringExtension.cs:11]: Anything else is considered as whitespace... oh wait, i forgot that this parser is not a hardened parser...

Comment (Arkatufus) [reply src/Akka.Bootstrap.Docker/StringExtension.cs:11]: I'll change that, give me a sec

Comment (Aaronontheweb) [reply src/Akka.Bootstrap.Docker/StringExtension.cs:11]: That value might actually be ok - tests passed on Linux and Windows. Add a test case to the suite that would hit this and see if it makes a difference

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**